graph/src/data/store/scalar.rs

```diff
@@ -71,20 +69,18 @@ impl BigDecimal {
         if self == &BigDecimal::zero() {
             return BigDecimal::zero();
         }
-        let (bigint, exp) = self.0.as_bigint_and_exponent();
+
+        // Round to the maximum significant digits.
+        let big_decimal = self.0.with_prec(Self::MAX_SIGNFICANT_DIGITS as u64);
```

Comment:
When switching to decimal128 we may need to remember that this method has a specific rounding mode and we need to match it or make a version bump.

Reply:
Hopefully there isn't too much variation in how to round a decimal number, but versioning might be the best anyways to make sure we don't miss any subtle behavior difference.